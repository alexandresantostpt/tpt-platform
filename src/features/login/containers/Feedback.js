import React from 'react'

import GlobalStyle from '@components/globalStyle/GlobalStyle'
import Toasts from '@components/toast/Toasts'

import Feedback from '../pages/Feedback'

const FeedbackContainer = () => (
    <>
        <GlobalStyle />
        <Toasts />
        <Feedback />
    </>
)

export default FeedbackContainer
