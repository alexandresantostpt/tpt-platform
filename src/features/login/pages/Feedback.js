import React, { useEffect } from 'react'
import styled from 'styled-components'

import { getQueryParams, navigateTo } from '@utils/browser'

import Text from '../components/text/Text'
import LoginComponent from '../components/login/Login'
import Icon from '@components/icon/Icon'

const StyledIcon = styled.span`
    align-items: center;
    background: linear-gradient(-225deg, #5271c4 0%, #b19fff 48%, #eca1fe 100%);
    border-radius: 50%;
    color: white;
    display: flex;
    height: 4.375rem;
    justify-content: center;
    width: 4.375rem;
`

const StyledBox = styled.div`
    display: flex;
    justify-content: center;
    margin-bottom: 1.75rem;
    margin-top: 1.75rem;
    width: 100%;
`

const StyledSpacing = styled.div`
    margin-bottom: 3rem;
`

const Feedback = () => {
    const { path, message } = getQueryParams()

    useEffect(() => {
        if (path) {
            setTimeout(() => navigateTo(path), 3000)
        }
    }, [])

    return (
        <LoginComponent>
            <StyledBox>
                <StyledIcon>
                    <Icon fontSize="4rem" icon="check" />
                </StyledIcon>
            </StyledBox>
            <Text>{message}</Text>
            <StyledSpacing />
        </LoginComponent>
    )
}

export default Feedback
