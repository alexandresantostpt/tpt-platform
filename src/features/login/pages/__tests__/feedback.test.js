import React from 'react'
import renderer from 'react-test-renderer'

import Feedback from '../Feedback'

describe('Tests for Feedback component', () => {
    it('Should render withou crash', () => {
        const tree = renderer.create(<Feedback />).toJSON()
        expect(tree).toMatchSnapshot()
    })
})
