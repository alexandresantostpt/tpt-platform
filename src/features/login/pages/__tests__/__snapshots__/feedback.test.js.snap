// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Feedback component Should render withou crash 1`] = `
<main
  className="Page__StyledPage-sc-9alblw-0 eBjaLI"
>
  <div
    className="Background__StyledBg-sc-904t04-0 bTqgFQ"
  />
  <div
    className="Box__StyledBox-v5mj6b-0 llPnpv"
  >
    <div
      className="Feedback__StyledBox-bpbx1j-1 jCUzAm"
    >
      <span
        className="Feedback__StyledIcon-bpbx1j-0 amCva"
      >
        <span
          className="icon icon-check Icon__StyledIcon-crr1cg-0 crXXNX"
          disabled={false}
          fontSize="4rem"
          onClick={null}
        />
      </span>
    </div>
    <div
      className="Text__StyledText-fa81pq-0 bIShHK"
    />
    <div
      className="Feedback__StyledSpacing-bpbx1j-2 dlROWv"
    />
  </div>
</main>
`;
